types.nix

```diff
@@ -235,9 +236,82 @@ rec {
         type = types.attrsOf lvm_vg;
         default = {};
       };
+      nodev = mkOption {
+        type = types.attrsOf nodev;
+        default = {};
+      };
     };
   };
 
+  nodev = types.submodule ({ config, ... }: {
+    options = {
+      type = mkOption {
+        type = types.enum [ "nodev" ];
+        default = "nodev";
+        internal = true;
+      };
+      fsType = mkOption {
+        type = types.enum [ "tmpfs" "ramfs" "nfs" ];
```

Comment:
do we want a str here? there are probably a lot of different fsTypes and we don't want to take care of all of them here. but for some we probably would need some extra packages.

Reply:
A string sounds reasonable.